crates/gitbutler-core/src/ops/entry.rs

```diff
@@ -17,16 +18,18 @@ use serde::Serialize;
 #[serde(rename_all = "camelCase")]
 pub struct Snapshot {
     /// The sha of the commit that represents the snapshot
-    pub id: String,
-    /// Snapshot creation time in epoch seconds
-    pub created_at: i64,
+    #[serde(with = "crate::serde::oid")]
+    pub id: git::Oid,
+    /// Snapshot creation time in seconds from Unix epoch seconds, based on a Git commit
+    #[serde(serialize_with = "crate::serde::as_time_seconds_from_unix_epoch")]
+    pub created_at: git2::Time,
```

Comment:
i really like this approach - it's clear in rust what it was, and we get the thing serialize correctly automatically :) 
Gonna copy this pattern in other places